src/components/dialogs/dynamicsimulation/event/dynamic-simulation-event-dialog.tsx

```diff
@@ -0,0 +1,179 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { FormProvider, useForm } from 'react-hook-form';
+import ModificationDialog from '../../commons/modificationDialog';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { FetchStatus } from '../../../../utils/rest-api';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useOpenShortWaitFetching } from '../../commons/handle-modification-form';
+import { FORM_LOADING_DELAY } from '../../../network/constants';
+import { DialogProps } from '@mui/material/Dialog/Dialog';
+import { DynamicSimulationEventForm } from './dynamic-simulation-event-form';
+import { Event, EventPropertyName, EventType } from './types/event.type';
+import yup from 'components/utils/yup-config';
+import { equipments } from '../../../network/network-equipments';
+import { getSchema } from './util/event-yup';
+import { eventDefinitions } from './model/event.model';
+import { saveEvent, getEvent } from '../../../../services/dynamic-simulation';
+
+export type DynamicSimulationEventDialogProps = {
+    studyUuid: string;
+    currentNodeId: string;
+    equipmentId: string;
+    equipmentType: string; // must be a string enum
+    isUpdate: boolean;
+    onClose: () => void;
+    editDataFetchStatus: string; // must be a string enum
+    title: string;
+} & DialogProps;
+
+const getEventType = (equipmentType: string): EventType | undefined => {
+    let eventType = undefined;
+    switch (equipmentType) {
+        case equipments.lines:
+        case equipments.twoWindingsTransformers:
+            eventType = EventType.DISCONNECT;
+            break;
+        default:
+    }
+
+    return eventType;
+};
+
+export const DynamicSimulationEventDialog = (
+    props: DynamicSimulationEventDialogProps
+) => {
+    const {
+        studyUuid,
+        currentNodeId,
+        equipmentId,
+        equipmentType,
+        isUpdate,
+        onClose,
+        editDataFetchStatus,
+        title,
+        open: defaultOpen,
+        ...dialogProps
+    } = props;
+
+    const [selectedId, setSelectedId] = useState(equipmentId ?? null);
+    const [dataFetchStatus, setDataFetchStatus] = useState(FetchStatus.IDLE);
+    const [event, setEvent] = useState<Event>();
+
+    const waitingOpen = useOpenShortWaitFetching({
+        isDataFetched: true,
+        delay: FORM_LOADING_DELAY,
+    });
+    const open = defaultOpen !== undefined ? defaultOpen : waitingOpen;
+
+    // empty form
+    const handleSetValuesAndEmptyOthers = useCallback(() => {}, []);
+
+    // submit form
+    const handleSubmit = useCallback((event: Event) => {
+        const eventWithId = { ...event, staticId: equipmentId };
+        saveEvent(studyUuid, currentNodeId, eventWithId)
+            .then((event) => {
+                console.log('Save successfully event : ', event);
+            })
+            .catch((error) => {
+                // should use snackError from useSnackMessage in common-ui
```

Comment:
TS not happy when importing useSnackMessage in common-ui

TS7016: Could not find a declaration file for module '@gridsuite/commons-ui'. '/home/phamquy/Projects/gridsuite/gridstudy-app/node_modules/@gridsuite/commons-ui/lib/index.js' implicitly has an 'any' type.   Try `npm i --save-dev @types/gridsuite__commons-ui` if it exists or add a new declaration (.d.ts) file containing `declare module '@gridsuite/commons-ui';`